### PR TITLE
Add an argument to ROUND func to show decimal numbers

### PR DESCRIPTION
**Pull request by Genne23v**
## Change Summary

Provide summary of changes with issue number if any.

This PR is to address [issue-3441](https://github.com/nocodb/nocodb/issues/3441). This change enables users to see decimal numbers when using `ROUND()` function. Also I updated usage description. 

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

This change is frontend only. I don't see any tests for frontend. Please let me know if any test exists for frontend. 

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

Screen Shot 2022-10-14 at 2 42 56 PM
Screen Shot 2022-10-14 at 2 43 05 PM

### CODE REVIEW

**Comment by o1lab** [comment]
@Genne23v - pls sign the CLA

**Comment by Genne23v** [comment]
@o1lab Do I need to do any action for the cypress failures? It seems nothing to do with my code. Please let me know. Thanks

**Comment by o1lab** [comment]
Will get back on it tomorrow. Thank you for the PR

**Comment by wingkwong** [comment]
> @o1lab Do I need to do any action for the cypress failures? It seems nothing to do with my code. Please let me know. Thanks

Actually Yes. Your changes break the existing logic.

**Review by wingkwong — CHANGES_REQUESTED** [review]
Since `precision` is allowed, please also help add a test case in `scripts/cypress/integration/common/3b_formula_column.js`

File: packages/nc-gui/utils/formulaUtils.ts [hunk 1]
```diff
@@ -176,12 +176,12 @@ const formulas: Record<string, any> = {
     type: formulaTypes.NUMERIC,
     validation: {
       args: {
-        rqd: 1,
+        rqd: 2,
```
[the remark this answers is not in the document]
Comment (Genne23v) [reply packages/nc-gui/utils/formulaUtils.ts:179]: I should've checked other functions too. Will be addressed. Accordingly.

File: packages/nc-gui/utils/formulaUtils.ts [hunk 2]
```diff
@@ -176,12 +176,12 @@ const formulas: Record<string, any> = {
...
       },
     },
-    description: 'Nearest integer to the input parameter',
-    syntax: 'ROUND(value)',
-    examples: ['ROUND(3.1415) => 3', 'ROUND({column1})'],
+    description: 'Rounded number to a specified number of decimal places',
+    syntax: 'ROUND(value, decimals)',
+    examples: ['ROUND(3.1415,2) => 3.14', 'ROUND({column1}, 3)'],
```
[the remark this answers is not in the document]
Comment (Genne23v) [reply packages/nc-gui/utils/formulaUtils.ts:184]: Will be addressed as well

**Comment by Genne23v** [comment]
@wingkwong I checked five SQL service providers if any of them doesn't support the decimal length for `ROUND` function. I've confirmed they all support decimal position. And I don't see server side `ROUND` function validation. I hope I'm not missing anything, but please let me know if some logic needs to be taken care of in backend. Thanks in advance!

**Comment by wingkwong** [comment]
I think it's fine. We can directly take the generated values.

**Comment by Genne23v** [comment]
Sorry that I needed some time to confirm the cypress integration test. It had old table from previous run, so my test was continuously failing. Now I have a green tag in all tests. Please review my change and let me know if you need anything. Thank you

File: packages/nc-gui/utils/formulaUtils.ts [hunk 3]
```diff
@@ -176,12 +176,13 @@ const formulas: Record<string, any> = {
...
-        rqd: 2,
+        min: 1,
+        max: 2,
       },
     },
-    description: 'Rounded number to a specified number of decimal places',
-    syntax: 'ROUND(value, decimals)',
-    examples: ['ROUND(3.1415,2) => 3.14', 'ROUND({column1}, 3)'],
+    description: 'Rounded number to a specified number of decimal places or the nearest integer if not specified',
+    syntax: 'ROUND(value, length), ROUND(value)',
```
Comment (wingkwong) [inline packages/nc-gui/utils/formulaUtils.ts:184]: I think it's better to call precision and we can just show the below one.

```
syntax: 'ROUND(value, precision)',
```

File: packages/nc-gui/utils/formulaUtils.ts [hunk 4]
```diff
@@ -176,12 +176,13 @@ const formulas: Record<string, any> = {
...
+    examples: ['ROUND(3.1415,2) => 3.14', 'ROUND(3.1415) => 3', 'ROUND({column1}, 3)'],
```
Comment (wingkwong) [inline packages/nc-gui/utils/formulaUtils.ts:185]: swap 1st and 2nd example to show the base case first.

```
examples: ['ROUND(3.1415) => 3', 'ROUND(3.1415, 2) => 3.14', 'ROUND({column1}, 3)'],
```

**Review by wingkwong — CHANGES_REQUESTED** [review]
Overall it looks good. Just a few minor changes.

Also, please update ROUND syntax and sample in `/nocodb/packages/noco-docs/content/en/setup-and-usages/formulas.md`.

https://github.com/nocodb/nocodb/blob/1e8c56a1c4eb547b8738323ed1d77ae34ef9be93/packages/noco-docs/content/en/setup-and-usages/formulas.md?plain=1#L52

File: packages/nc-gui/utils/formulaUtils.ts (hunk 3 above)
Comment (Genne23v) [reply packages/nc-gui/utils/formulaUtils.ts:184]: Thanks for the quick reply. I will push the update soon

File: packages/nc-gui/utils/formulaUtils.ts [hunk 5]
```diff
@@ -181,8 +181,8 @@ const formulas: Record<string, any> = {
       },
     },
     description: 'Rounded number to a specified number of decimal places or the nearest integer if not specified',
-    syntax: 'ROUND(value, length), ROUND(value)',
-    examples: ['ROUND(3.1415,2) => 3.14', 'ROUND(3.1415) => 3', 'ROUND({column1}, 3)'],
+    syntax: 'ROUND(value, precision), ROUND(value)',
+    examples: ['ROUND(3.1415) => 3', 'ROUND(3.1415,2) => 3.14', 'ROUND({column1}, 3)'],
```
Comment (wingkwong) [inline packages/nc-gui/utils/formulaUtils.ts:185]: Prefer a space to make it consistent, i.e. `ROUND(3.1415, 2)`

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Review by wingkwong — APPROVED** [review]
LGTM. The tests are still running. Let's wait for the results.

**Comment by wingkwong** [comment]
I'll merge it. Tests failed due to LTAR issues which are not related to this PR.

**Comment by Genne23v** [comment]
> 

Thank you so much! It's really nice experience for me to contribute to nocodb!

**Pull request merged**